### ISSUE TITLE
variable name changes in `PublicUserData` type for org membership response

### ISSUE THREAD

**Issue by dalalRohit** [clerk/javascript#4775]
### Preliminary Checks

- [X] I have reviewed the documentation: https://clerk.com/docs

- [X] I have searched for existing issues: https://github.com/clerk/javascript/issues

- [X] I have not already reached out to Clerk support via email or Discord (if you have, no need to open an issue here)

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Clerk. Please ask questions in our Discord community: https://clerk.com/discord.

### Reproduction

https://github.com/dalalRohit/clerk-members-issue

### Publishable key

pk_test_d2VsbC1raXQtMzEuY2xlcmsuYWNjb3VudHMuZGV2JA

### Description

**Steps to reproduce:**

1. Clone the [repo](https://github.com/dalalRohit/clerk-members-issue) or visit the [reproduction](https://clerk-members-issue.vercel.app/)
2. Install dependencies
3. Run `npm run dev` to start app on `localhost:3000`
4. Sign-in with Github/Google account
5. Create a new organisation from the home-page (Create Org Link) & add members.
6. On the home page when I display the data, the user data is with _snake_case_. But the actual data logged into console is _camelCase_. 
You can see the difference in the case (_camelCase_ vs _snake_case_) in the memberships data of the current organisation which is fetched via `useOrganization()` hook
<img width="1438" alt="image" src="https://github.com/user-attachments/assets/eacd8ec5-3922-48bf-aeac-a8f382cf4c38" />

This sudden & unexpected change of API response has caused our production app to go down and it is not yet resolved. 

As we are trying to read `publicUserData` from individual `member` from the `membersList`. But, `public_user_data` is available and not `publicUserData`. Thus getting `null` everywhere for the membership entries.
![actual code snippet from our app](https://github.com/user-attachments/assets/a819a704-399d-4674-974e-d4bc10906677)

This change has happened suddenly overnight and [as per docs](https://clerk.com/docs/references/javascript/types/public-user-data) the response structure has not changed. Kindly please help in resolving this issue. 

Any help highly appreciated. Thank you.

### Environment

```Shell
System:
    OS: macOS 13.4
    CPU: (8) arm64 Apple M1
    Memory: 174.13 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
Binaries:
    Node: 20.12.1 - ~/.nvm/versions/node/v20.12.1/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 10.8.2 - ~/.nvm/versions/node/v20.12.1/bin/npm
Browsers:
    Brave Browser: 131.1.73.91
    Chrome: 131.0.6778.140
    Safari: 16.5
npmPackages:
    @clerk/nextjs: ^6.1.0 => 6.1.0 
    @types/node: ^22.8.6 => 22.8.6 
    @types/react: ^18.3.12 => 18.3.12 
    @types/react-dom: ^18.3.1 => 18.3.1 
    autoprefixer: ^10.4.20 => 10.4.20 
    eslint: ^9.13.0 => 9.13.0 
    eslint-config-next: ^15.0.2 => 15.0.2 
    next: ^15.0.2 => 15.0.2 
    postcss: ^8.4.47 => 8.4.47 
    react: ^18.3.1 => 18.3.1 
    react-dom: ^18.3.1 => 18.3.1 
    tailwindcss: ^3.4.14 => 3.4.14 
    typescript: ^5.6.3 => 5.6.3
```

**Comment by panteliselef** [comment]
@dalalRohit thanks for reporting this issue.

Can you share with us your real world use case for using JSON.stringify with Clerk resources ?

**Comment by anagstef** [comment]
Update on this: The issue has been resolved!

Thank you for reporting and sorry for all the inconvenience caused by this. 😞

**Comment by dalalRohit** [comment]
Thank you so much for the quick resolution of the above issue. Really appreciate it 🙏🏻🙌🏻
Being curios, if you guys don't mind, can you please share what was the root cause over here?

**Comment by anagstef** [comment]
Hello @dalalRohit! 

We introduced a `toJSON()` method on our resources, and this method is called internally by `JSON.stringify` if it's defined. So, after introducing it, the `JSON.stringify` returns a slightly different structure (snake_case instead of camelCase) on the properties and this was a breaking change for anyone serializing our resources.

We reverted this and the issue was fixed!

Sorry for the inconvenience caused by this! 😞
